### PR TITLE
[UPCOMING]  Release 1.7.0 - workshop release

### PR DESCRIPTION
**Pull request by skoudoro**
Preparation for 1.7.0 release, targeting April 7, 2023 or April 14 2023. 

# Desired PR / issues (need review / feedback / update)

## PR

- [ ]  #2759  (Submitter: @frheault)
- [x] #2751 (Submitter: @CHrlS98 )
- [x] #2749 (Submitter: @shilpiprd)
- [x] #2744 (Submitter: @guaje )
- [x] #2740 (Submitter: @RahulUbale)
- [x] #2626 (Submitter: @pjsjongsung)
- [x] #2596 (Submitter: @gabknight )
- [x] #2540 (Submitter: @BramshQamar  )
- [x] #2762 (Submitter: @skoudoro)
- [x]  #2768 (Submitter: @BramshQamar  )
- [x]  #2773 (Submitter: @pjsjongsung)
- [x] #2782 (Submitter: @skoudoro)
- [x] #2609 (Submitter: @tomdelahaije )
- [x] #2788(Submitter: @skoudoro)
- [x] #2765(Submitter: @skoudoro)

## Issues

- [ ] #2736 
- [ ] #2706
- [ ] #2671 
- [ ] #2648 
- [x] #2530

## Feature without PR yet:

- ~~Masked P2S Algorithm~~. (will not be added)
- ~~NUQ Algorithm~~. (will not be added)

Most of them can be done on time, I think :smile:

Please comment to flag any other issues or PR that should be addressed for this release

# Release checklist
- [x] add release_notes 1.6.0
- [x] Update changelog and api_changes
- [x] Update .mailmap
- [x] Set release number in info.py
- [x] Check the copyright years in doc/conf.py and LICENSE
- [x] Check documentation
- [x] Check Tutorial
- [x] Check unit test and USE-PRE
- [ ] Update index.rst
- [ ] Update Highligth.rst, old_news.rst, stateofheart.rst
- [x] Update developers.rst list
- [ ] Update website
- [x] Update deprecation cycle remove long deprecation

### CODE REVIEW

File: Changelog [hunk 1]
```diff
@@ -24,6 +24,20 @@ Dipy
 
 The code found in Dipy was created by the people found in the AUTHOR file.
 
+* 1.7.0 (Sunday, April 23, 2023)
+
+- NF: BundleWarp - Streamline-based Nonlinear registration added.
+- NF: DKI+ - Diffusion Kurtosis constraints added.
+- NF: Synb0 added.
```
Comment (Garyfallidis) [inline Changelog:31]: Synb0 - synthetic b0 creation added using deep learning

File: Changelog [hunk 2]
```diff
@@ -24,6 +24,20 @@ Dipy
 
 The code found in Dipy was created by the people found in the AUTHOR file.
 
+* 1.7.0 (Sunday, April 23, 2023)
+
+- NF: BundleWarp - Streamline-based Nonlinear registration added.
+- NF: DKI+ - Diffusion Kurtosis constraints added.
```
Comment (Garyfallidis) [inline Changelog:30]: Diffusion Kurtosis modeling with advanced constraints

File: Changelog [hunk 3]
```diff
@@ -24,6 +24,20 @@ Dipy
 
 The code found in Dipy was created by the people found in the AUTHOR file.
 
+* 1.7.0 (Sunday, April 23, 2023)
+
+- NF: BundleWarp - Streamline-based Nonlinear registration added.
```
Comment (Garyfallidis) [inline Changelog:29]: Streamline-based nonlinear registration method for bundles

**Review by Garyfallidis — CHANGES_REQUESTED** [review]

**Comment by skoudoro** [comment]
all good ! merging and starting to tag the release.

**Pull request merged**